File: backend/Signal/src/main/java/com/ssafy/signal/board/domain/BoardDto.java
package com.ssafy.signal.board.domain;

import com.ssafy.signal.file.domain.FileDto;
import com.ssafy.signal.member.dto.findMemberDto;
import lombok.*;
import java.time.LocalDateTime;
import java.util.*;
import java.util.stream.Collectors;

import com.ssafy.signal.member.domain.Member;

@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class BoardDto {
    private Long id;
    private String writer;
    private Long userId; // 새로운 필드 추가
    private findMemberDto member;
    private String title;
    private String content;
    private Long reference;
    private Long liked;
    private Long type;
    private LocalDateTime createdDate;
    private LocalDateTime modifiedDate;
    private List<CommentDto> comments;
    private List<TagDto> tags;
    private List<String> fileUrls;



    public BoardEntity toEntity(){
        return  BoardEntity.builder()
                .id(id)
                .writer(writer)
                .title(title)
                .content(content)
                .reference(reference)
                .liked(liked)
                .type(type)
                .tags(tags.stream().map(TagDto::toEntity).toList())
                .user(Member.builder().userId(userId).build()) // Member 객체 참조
                .build();
    }

    public BoardEntity toEntity(List<CommentDto> comments){
        return  BoardEntity.builder()
                .id(id)
                .writer(writer)
                .title(title)
                .content(content)
                .reference(reference)
                .liked(liked)
                .type(type)
                .tags(tags.stream().map(TagDto::toEntity).toList())
                .user(Member.builder().userId(userId).build()) // Member 객체 참조
                .build();
    }
}

File: android/Signal/app/src/main/java/com/ongo/signal/data/model/main/BoardDTO.kt
package com.ongo.signal.data.model.main

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

data class BoardDTO(
    val comments: ArrayList<CommentDTOItem>? = arrayListOf(),
    val content: String,
    val createdDate: String? = null,
    val id: Long = 0,
    val liked: Long = 0,
    val modifiedDate: String? = null,
    val reference: Long = 0,
    val title: String,
    val type: Long? = 0,
    val userId: Long,
    val writer: String,
    @SerializedName("fileUrls") val imageUrls: List<String>? = listOf(),
    val tags: List<TagDTO>? = emptyList()
) {
    fun getFormattedCreatedDate(): String? {
        return createdDate?.let { formatDate(it) }
    }

    fun getFormattedModifiedDate(): String? {
        return modifiedDate?.let { formatDate(it) }
    }

    private fun formatDate(dateTime: String): String {
        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
        val parsedDate = LocalDateTime.parse(dateTime)
        return parsedDate.format(formatter)
    }

    fun getCommentCount(): Int {
        return comments?.size ?: 0
    }
}